### DISCUSSION TITLE
Some Questions About UseEventEmitter Hook

### DISCUSSION THREAD

**Discussion by KeithChou** [alibaba/hooks#1558] [General]
![image](https://user-images.githubusercontent.com/18112461/163667300-da950e85-ebd2-4e5e-b7e9-6a0378b2d3b1.png)
The above pic is the origin code about useEventEmitter. There are some questions about it.
I will appreciate it if u guys can response😃😃😃.

1. the useEventEmitter hooks take use of // eslint-disable-next-line react-hooks/rules-of-hooks to ignore the eslint check, which may discrupt the call sequence of react's hooks. Such as the following writing might cause problems .
![image](https://user-images.githubusercontent.com/18112461/163667754-063abfdb-7cdd-4755-a3c8-0ceb6730f80f.png)
In general, we use useEventEmitter in async method or condition , is it will discrupt the call sequence of react's hooks?    
2. Why not use the corresponding name? that is, useSubscription and useEmit instead of useSubscription and emit?

**Comment by brickspert** [comment]
1. Eslint's warnings can be ignored if you know what you're doing. We know for sure that the Hooks rule is not going to be broken here.
Then, you can use `emit` in async method or condition, not `useEventEmitter`.

2. `emit` is not  Hooks，so cannot be `useEmit`.

**Reply by KeithChou** [reply]
Very happy to receive ur reply 🤓🤓🤓.

1. According to your answer for the first point,  i will give u that , but  u  cant decide how the developers useEventEmitter or focus$.useSubscription . There is a situation that they use them in async function of condition, it will broke there application ,and its diffcult for us to position issues. I dont know if u agree with me. : (
In my opinion, Using this methods require developer understand and use correctly. There occurs bad positioning issues while they wrong use, but there no errors or warnings were thrown . 
U know what i meanings? There are different level of developers in the word .. : (
2. For the second point, this explanation does not solve my doubts, i am so sorry for that. Further More, why subscribe event is a hooks and emit event is not??

Look forward to ur answers and replies 👏👏

**Reply by brickspert** [reply]
1. React provide `useState`, and he can't decide how you use it. If you use it incorrectly, you will report an error.  All Hooks is same to `useState`.
2. I don't know how to answer you, maybe you should know the difference between functions and Hooks.

**Reply by KeithChou** [reply]
yep . u are right . But in the useEventEmitter u disable the eslint check which cause eslint can't throw an error for us🤔 .

for the second one, .i know the difference between . but there are a thousand Hamlets for a thousand readers😀
